lldb/source/Plugins/Language/Swift/SwiftFormatters.cpp

```diff
@@ -736,6 +741,104 @@ class EnumSyntheticFrontEnd : public SyntheticChildrenFrontEnd {
   ConstString m_element_name;
   size_t m_child_index;
 };
+
+class TaskSyntheticFrontEnd : public SyntheticChildrenFrontEnd {
+public:
+  TaskSyntheticFrontEnd(lldb::ValueObjectSP valobj_sp)
+      : SyntheticChildrenFrontEnd(*valobj_sp.get()) {}
+
+  constexpr static StringLiteral TaskChildren[] = {
+      "isChildTask",    "isFuture",    "isGroupChildTask",
+      "isAsyncLetTask", "isCancelled", "isStatusRecordLocked",
+      "isEscalated",    "isEnqueued",  "isRunning",
+  };
+
+  llvm::Expected<uint32_t> CalculateNumChildren() override {
+    auto count = ArrayRef(TaskChildren).size();
+    return m_task_info.hasIsRunning ? count : count - 1;
+  }
+
+  lldb::ValueObjectSP GetChildAtIndex(uint32_t idx) override {
+    auto target_sp = m_backend.GetTargetSP();
+#define RETURN_CHILD(FIELD, NAME)                                              \
+  if (!FIELD)                                                                  \
+    FIELD = ValueObject::CreateValueObjectFromBool(target_sp,                  \
+                                                   m_task_info.NAME, #NAME);   \
+  return FIELD
+
+    switch (idx) {
+    case 0:
+      RETURN_CHILD(m_is_child_task_sp, isChildTask);
+    case 1:
+      RETURN_CHILD(m_is_future_sp, isFuture);
+    case 2:
+      RETURN_CHILD(m_is_group_child_task_sp, isGroupChildTask);
+    case 3:
+      RETURN_CHILD(m_is_async_let_task_sp, isAsyncLetTask);
+    case 4:
+      RETURN_CHILD(m_is_cancelled_sp, isCancelled);
+    case 5:
+      RETURN_CHILD(m_is_status_record_locked_sp, isStatusRecordLocked);
+    case 6:
+      RETURN_CHILD(m_is_escalated_sp, isEscalated);
+    case 7:
+      RETURN_CHILD(m_is_enqueued_sp, isEnqueued);
+    case 8:
+      RETURN_CHILD(m_is_running_sp, isRunning);
+    default:
+      return {};
+    }
+
+#undef RETURN_CHILD
+  }
+
+  lldb::ChildCacheState Update() override {
+    if (auto *runtime = SwiftLanguageRuntime::Get(m_backend.GetProcessSP())) {
+      auto reflection_ctx = runtime->GetReflectionContext();
+      auto task_obj_sp = m_backend.GetChildMemberWithName("_task");
+      auto task_ptr = task_obj_sp->GetValueAsUnsigned(LLDB_INVALID_ADDRESS);
```

Comment:
generally I would go for more explicit types over auto if the types aren't making the code unreadable.